server/lib/reverse_etl/extractors/incremental_delta.rb

```diff
@@ -40,6 +40,8 @@ def process_record(message, sync_run, model)
         fingerprint = generate_fingerprint(record.data)
         primary_key = record.data.with_indifferent_access[model.primary_key]
 
+        return if primary_key.blank?
```

Comment:
Instead of just skipping the record. We should throw an error which states that the primary key is missing. Else it will be difficult to debug what caused the issue.